native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe.Tests/deserializer/avro/AvroDeserializerTest.cs

```diff
@@ -0,0 +1,536 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using Avro;
+using Avro.Generic;
+using Avro.IO;
+using AWSGsrSerDe.common;
+using AWSGsrSerDe.deserializer.avro;
+using AWSGsrSerDe.serializer.avro;
+using AWSGsrSerDe.Tests.utils;
+using NUnit.Framework;
+using Encoder = Avro.IO.Encoder;
+
+namespace AWSGsrSerDe.Tests.deserializer.avro
+{
+    [TestFixture]
+    public class AvroDeserializerTest
+    {
+        private const string TestAvroSchema = "{\"namespace\": \"AWSGsrSerDe.Tests\",\n"
+                                              + " \"type\": \"record\",\n"
+                                              + " \"name\": \"User\",\n"
+                                              + " \"fields\": [\n"
+                                              + "     {\"name\": \"name\", \"type\": \"string\"},\n"
+                                              + "     {\"name\": \"favorite_number\",  \"type\": [\"int\", \"null\"]},\n"
+                                              + "     {\"name\": \"favorite_color\", \"type\": [\"string\", \"null\"]}\n"
+                                              + " ]\n"
+                                              + "}";
+
+        private const string RelativePathPrefix = "../../../../../../../serializer-deserializer/src/test/";
+        private const string AvroEnumSchema = RelativePathPrefix+"resources/avro/user_enum.avsc";
+        private const string AvroArraySchema = RelativePathPrefix+"resources/avro/user_array.avsc";
+        private const string AvroStringArraySchema = RelativePathPrefix+"resources/avro/user_array_String.avsc";
+        private const string AvroUnionSchema = RelativePathPrefix+"resources/avro/user_union.avsc";
+        private const string AvroFixedSchema = RelativePathPrefix+"resources/avro/user_fixed.avsc";
+        private const string AvroMapSchema = RelativePathPrefix+"resources/avro/user_map.avsc";
+        private const string AvroMixedTypeSchema = RelativePathPrefix+"resources/avro/user3.avsc";
```

Comment:
Is there a way read these as files? We can reuse that utility for Protobuf and JSON tests and avoid repetition.

Reply:
I made a update with a different approach